pyOpenSci/pyopensci.github.io
# Fix Packages Page Listing Missing Maintainer

# What was done
---
Addressing issue https://github.com/pyOpenSci/pyopensci.github.io/issues/169

- Updated the _packages.yml_ file with latest package data
- Modified _package-grid.html_ template to conditionally display contributor names in the following manner
-- If the `all_current_maintainers` value is present for the package in the Yaml file, loop through the maintainers' names (with fallback to GitHub user names) and display in a comma-delimited format
-- Otherwise, display the submitting author's name (with fallback to GitHub user name)
- Create a link on the displayed user name to each maintainer's public GitHub profile
- Removed extraneous _packages.yml_ and _contributors.yml_ files from application root

See the following screen clipping example:
<img width="1154" alt="Screen Shot 2023-04-27 at 10 11 53 PM" src="https://user-images.githubusercontent.com/6424805/235040215-37e39282-befb-43e2-b071-0611a3d7ccd0.png">

# How to test
---
- Open the _python-packages_ template in the browser
- Note that packages with multiple contributors now display each contributor in a comma-separated list
- Note that packages with missing "author" names now display the GitHub username of the submitting author
- Observe that, by clicking author user names, the public GitHub profile page is opened (in a new tab/window) for that maintainer

---
Special thanks to the team at PyCon 2023 for making this possible, with their help and encouragement! Thank you especially to @lwasser.
